### PR TITLE
7788 read only skip rule

### PR DESCRIPTION
**Pull request by DevneyHamilton**
Branched off of  7178. 
screen shot 2018-03-07 at 10 08 07 am

### CODE REVIEW

File: app/assets/stylesheets/all/components/_forms.scss [hunk 1]
```diff
@@ -19,6 +19,19 @@ form.form_form {
   }
 }
 
+// Override bootstrap on dl's and dt's for read only questioning
+dt {
```
Comment (DevneyHamilton) [inline app/assets/stylesheets/all/components/_forms.scss]: don't know best place to put this css stuff

File: app/decorators/skip_rule_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  def display_dest
+    if destination == "item"
+      "#{I18n.t("activerecord.models.question.one")} #{QuestioningDecorator.new(dest_item).name_and_rank}"
+    else
+      I18n.t("skip_rule.end_of_form")
+    end
+  end
+
+  def read_only_header
+    "#{I18n.t("skip_rule.skip_to_item", label: display_dest)} #{I18n.t("skip_rule.skip_if_options.#{skip_if}")}"
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply app/decorators/skip_rule_decorator.rb]: not sure about this for i18n - please advise.

[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply app/decorators/skip_rule_decorator.rb]: should I make separate strings for the destination and the skip if parts of this?

File: app/assets/stylesheets/all/components/_forms.scss (hunk 1 above)
Comment (hooverlunch) [reply app/assets/stylesheets/all/components/_forms.scss]: I think it's fine in here but I'd put a generic comment like "dl can be used for non bulleted lists of things in a read only form view. These styles put 20px of space between each dl, but not on either end."

File: app/assets/stylesheets/all/components/_forms.scss [hunk 3]
```diff
@@ -19,6 +19,19 @@ form.form_form {
...
+  margin-top: 20px;
+}
+
+dt:first-of-type {
```
Comment (hooverlunch) [inline app/assets/stylesheets/all/components/_forms.scss]: These style should be on `dl` (the outer tag), not `dt`. I'm not sure how this is working currently.

File: app/decorators/skip_rule_decorator.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (hooverlunch) [reply app/decorators/skip_rule_decorator.rb]: I think nested interpolations are bit hard on the eyes... Could you assign the I18n results to vars and then interpolate them together?

[the remark this answers is not in the document]
Comment (hooverlunch) [reply app/decorators/skip_rule_decorator.rb]: Another option would be to concat them together... so `I18n.t(...) << " " << I18n.t(...)`

File: app/decorators/skip_rule_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+class SkipRuleDecorator < ApplicationDecorator
+  delegate_all
+
+  def human_readable
+    I18n.t("skip_rule.instructions",
+      destination: display_dest,
+      conditions: decorate_conditions)
+  end
+
+  def decorate_conditions
```
Comment (hooverlunch) [inline app/decorators/skip_rule_decorator.rb]: Do all the below methods need to be public? Default should be private, only public if they need to be called from outside.

File: app/views/form_items/_printable.html.erb [hunk 5]
```diff
@@ -5,6 +5,6 @@
       <%= render 'form_items/printable', items: subitems %>
     <% end %>
   <% elsif item.printable? %>
-    <%= render 'questionings/printable', qing: Printable::QingDecorator.new(item) %>
+    <%= render 'questionings/printable', qing: QuestioningDecorator.new(item) %>
```
Comment (hooverlunch) [inline app/views/form_items/_printable.html.erb]: Change to dbl quotes plz (no hound for ERB yet :P)

File: app/views/questionings/_skip_logic.html.erb [hunk 6]
```diff
@@ -1,4 +1,13 @@
 <% if read_only || cannot?(:update_core, @questioning) %>
+    <dl>
+
```
Comment (hooverlunch) [inline app/views/questionings/_skip_logic.html.erb]: Remove extra line plz.

File: app/views/questionings/_skip_logic.html.erb [hunk 7]
```diff
@@ -1,4 +1,13 @@
...
+      <% @questioning.skip_rules.each do |rule| %>
+        <dt><%= SkipRuleDecorator.new(rule).read_only_header %></dt>
+        <% rule.condition_group.members.each do |c| %>
+          <dd><%= c.decorate.human_readable  %></dd>
```
Comment (hooverlunch) [inline app/views/questionings/_skip_logic.html.erb]: Remove extra space inside %> plz.

File: spec/features/skip_rule_form_spec.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                                     {ref_qing_id: form.c[1].id, op: "eq", value: "10"}
+                                   ])
+      expected = "Skip to #{dest_qing_str} if any of these conditions are met "\
+        "#{first_cond_str} "\
+        "#{second_cond_str}"
+      visit("/en/m/#{form.mission.compact_name}/questionings/#{form.c[2].id}")
+      expect(page).to have_content(expected)
+    end
+  end
+end
```
Comment (hooverlunch) [inline spec/features/skip_rule_form_spec.rb]: Nice spec!

File: app/assets/stylesheets/all/components/_forms.scss (hunk 3 above)
Comment (DevneyHamilton) [reply app/assets/stylesheets/all/components/_forms.scss]: Let's talk about this Fri am. <dl> wraps around all the skip rules, and the idea is to put space after each skip rule. Each skip rule is in a <dt>

**Pull request merged**